compiler/rustc_arena/src/lib.rs

```diff
@@ -214,7 +214,9 @@ impl<T> TypedArena<T> {
             // We need the element size to convert chunk sizes (ranging from
             // PAGE to HUGE_PAGE bytes) to element counts.
             let elem_size = cmp::max(1, mem::size_of::<T>());
-            let mut chunks = self.chunks.borrow_mut();
+
+            let chunks = &mut *self.chunks.as_ptr();
```

Comment:
Are you sure this is safe? `&self` allows for multiple shared borrows at once. If we're going to be getting references anyway I'd rather use refcell which at least does runtime checking instead of silent UB.

Reply:
I think we can't grow an arena multiple times at the same time, since it's not Sync and grow isn't recursive.

Reply:
I'm not sure if `grow` can panic, though, or just aborts when out of memory.